analysis/monk/src/DampenHarm.tsx

```diff
@@ -0,0 +1,73 @@
+import { formatNumber, formatThousands } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+class DampenHarm extends Analyzer {
+  hitsReduced = 0;
+  damageReduced = 0;
+  skip = 0;
+  currentMaxHP = 0;
+
+  constructor(options: Options) {
+    super(options);
+    if (!this.selectedCombatant.hasTalent(SPELLS.DAMPEN_HARM_TALENT.id)) {
+      this.active = false;
+      return;
+    }
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.updateMaxHP);
+    this.addEventListener(Events.resourcechange.by(SELECTED_PLAYER), this.updateMaxHP);
+    this.addEventListener(Events.drain.by(SELECTED_PLAYER), this.updateMaxHP);
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.damageReduction);
+  }
+
+  damageReduction(event: DamageEvent) {
+    if (event.ability.guid === SPELLS.STAGGER_TAKEN.id) {
+      return;
+    }
+    if (this.selectedCombatant.hasBuff(SPELLS.DAMPEN_HARM_TALENT.id)) {
+      this.hitsReduced += 1;
+      const h = event.amount || 0;
+      const a = event.absorbed || 0;
+      const o = event.overkill || 0;
+      // TODO: ACTUAL MAX HP TRACKING
+      // event.maxHitPoints is insufficient as many events such as SPELL_ABSORBED are missing it and Keg of the Heavens is cursed.
+      const maxHP = event.maxHitPoints || this.currentMaxHP;
```

Comment:
maxHP has a chance to equal 0 if the first damage event to the player is fully absorbed (and.. the player didn't receive any damage or healing before hand)
A simple `if (maxHP === 0) {return;}` will catch this edge case. 

It might seem minor or unlikely to happen but right when you don't code an edge case it decides to poke its nasty head out and fuck your code up

Reply:
Done.